### PR TITLE
Move our OOP syncing code off of source-text checksums, onto content hashes.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1880936
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1388754

Followup to https://github.com/dotnet/roslyn/pull/70838

### CODE REVIEW

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski @ToddGrun ptal :)

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker_Generators.cs [hunk 1]
```diff
@@ -188,16 +188,16 @@ private partial class CompilationTracker : ICompilationTracker
                         // Server provided us the checksum, so we just pass that along.  Note: it is critical that we do
                         // this as it may not be possible to reconstruct the same checksum the server produced due to
                         // the lossy nature of source texts.  See comment on GetOriginalSourceTextChecksum for more detail.
-                        contentIdentity.OriginalSourceTextChecksum);
-                    Contract.ThrowIfTrue(generatedDocument.GetOriginalSourceTextChecksum() != contentIdentity.OriginalSourceTextChecksum, "Checksums must match!");
+                        contentIdentity.OriginalSourceTextContentHash);
+                    Contract.ThrowIfTrue(generatedDocument.GetOriginalSourceTextContentHash() != contentIdentity.OriginalSourceTextContentHash, "Checksums must match!");
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker_Generators.cs:192]: >Checksums [](http://example.com/codeflow?start=145&length=9)

nit: consider updating comment

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker_Generators.cs:192]: nm, I got confused I guess

**Pull request merged**